docs/en/ingest-management/fleet/fleet-settings.asciidoc

```diff
@@ -239,3 +239,31 @@ data if no other output is set in the agent policy.
 
 |===
 
+[discrete]
+[[fleet-agent-binary-download-settings]]
+== Agent binary download settings
+
+{agent}s must be able to access the {artifact-registry} to download
+binaries during upgrades. By default {agent}s download artifacts from the
+artifact registry at `https://artifacts.elastic.co/downloads/`.
+
+For {agent}s that cannot access the internet, you can specify agent binary
+download settings, and then configure agents to download their artifacts from
+the alternate location. For more information about running {agent}s in a
+restricted environment, refer to <<air-gapped>>.
+
+To add or edit the source of binary downloads:
+
+. Click **Add agent binary source** or **Edit**.
+. Set the agent binary source name.
+. For **Host**, specify the address where you are hosting the artifacts
+repository.
+//REVIEWERS: The text in the UI describes the Host setting as follows, but to me,
```

Comment:
@michalpristas Did you see my question here about the host setting? The UI makes it sound like the users has to specify the full path to their binary files. That would be a lot of work, so I think maybe the UI is wrong, but wanted to confirm.

Reply:
it should not be path to binary but rather a tree structure as we discussed offline some time ago.

Reply:
Right...so the UI text is confusing because it suggests a single binary. I've opened an issue to track this here: https://github.com/elastic/kibana/issues/139151